cmd/nexusUpload.go

```diff
@@ -0,0 +1,138 @@
+package cmd
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"hash"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	"crypto/md5"
+	"crypto/sha1"
+
+	piperHttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+)
+
+type artifactDescription struct {
+	ID         string `json:"artifactId"`
+	Classifier string `json:"classifier"`
+	Type       string `json:"type"`
+	File       string `json:"file"`
+}
+
+func nexusUpload(config nexusUploadOptions, telemetryData *telemetry.CustomData) error {
```

Comment:
do not return an error here.
The step framework does not process the error anymore ...

Reply:
>The step framework does not process the error anymore ...

Wouldn't it be a benefit to let errors bubble up into the framework in order to establish some common way of error handling?

Reply:
I want to leave this up to you guys. My observation is that the code has become much cleaner without the manual error returning, i.e. when you just exit (via the logger). I have to say I agree with Marcus that I also do not think it is nice to exit via the logging framework, but this could be easily fixed by introducing a central exit function (which can then also log the error besides exiting). It is then up to the individual command to decide what errors it wants to recover from or ignore and has to bubble up to a certain level. But making it a requirement to bubble the error up to the top-level will certainly cause more bloat in the code. That's just my first impression of coding in go. (I don't understand why go has no exceptions... the reasons given in various blog posts don't convince me. But it is possible to install panic handlers, for what it is worth, and deferred functions will also run in case of panics.)